### Task1:
You need to code a program that prints numbers from 1 to 20 except multiple of 2 & 3.

Hint: use "modulo" operator and "and/or" operators

In [7]:
# Your code goes here
for num in range (1,21) :
    if (num%2==0) or (num%3==0):
        continue
    print (num)

1
5
7
11
13
17
19


### Task2:
Write a program that keep on accepting number from the user until user enters Zero. Display the sum and average of all the numbers.

Hint: Use while loop with flag. 

In [22]:
# Your code goes here
sum_n = 0
count = 0
flag = True
while flag:
    num = int(input("Enter any Number or press 0 to exit: "))
    if num == 0:
        flag =False
    else:
        sum_n += num
        count += 1
average = sum_n // count
print("the Sum of all the enterd number is ",sum_n)
print("the average of all the enterd number is",average)

Enter any Number or press 0 to exit: 25
Enter any Number or press 0 to exit: 26
Enter any Number or press 0 to exit: 27
Enter any Number or press 0 to exit: 22
Enter any Number or press 0 to exit: 0
the Sum of all the enterd number is  100
the average of all the enterd number is 25


### Task3:
Accept 10 numbers from the user and display their average.

Hint: Use loop for accepting numbers or use split to take input in one go

In [26]:
# Your code goes here
total = 0
count = 0
for i in range(10):
    num = int(input("Enter Any Number: "))
    total += num
    count += 1
average = total // count
print("The average of all enterd number is",average)

Enter Any Number: 24
Enter Any Number: 25
Enter Any Number: 26
Enter Any Number: 24
Enter Any Number: 26
Enter Any Number: 27
Enter Any Number: 24
Enter Any Number: 26
Enter Any Number: 27
Enter Any Number: 29
The average of all enterd number is 25


### Task4:
Write a program to display sum of odd numbers and even numbers that fall between 12 and 37(including both numbers)


In [1]:
# Your code goes here
sum_even = 0
sum_odd = 0
for i in range(12,38):
    if i%2==0:
        sum_even += i
    else:
        sum_odd += i
print("The Sum of Even Number From 12 to 37 is",sum_even)
print("The Sum of Odd Number From 12 to 37 is",sum_odd)

The Sum of Even Number From 12 to 37 is 312
The Sum of Odd Number From 12 to 37 is 325


### Task5:
Write a program to print table of a number accepted from user.


In [9]:
# Your code goes here
num = int(input("Enter a number: "))

print("Multiplication Table of", num)
for n in range(1, 11):
    print(num,"X",n,"=",num * n)

Enter a number: 7
Multiplication Table of 7
7 X 1 = 7
7 X 2 = 14
7 X 3 = 21
7 X 4 = 28
7 X 5 = 35
7 X 6 = 42
7 X 7 = 49
7 X 8 = 56
7 X 9 = 63
7 X 10 = 70


### Task6:
Write a program to display product of the digits of a number accepted from the user.

If user inputs:123 Outputs 6

If user inputs: 567 Outputs 210

In [3]:
# Your code goes here
num = int(input("Enter any number : "))

temp = num
product = 1;

while(temp != 0):

    product = product * (temp % 10);
    # Remove last digit from temp.
    temp = int(temp / 10)

print("Product of all digits in", num, "is", product)

Enter any number : 2545
Product of all digits in 2545 is 200


### Task7: 
Write a program to print first 10 even numbers in reverse order.

In [21]:
# Your code goes here
for a in range(20,0,-1):
    if a%2==0:
        print(a)
    else:
        pass

20
18
16
14
12
10
8
6
4
2


### Task8:
Write a program to find the sum of the digits of a number accepted from user

If user inputs:123 Outputs 6

If user inputs: 567 Outputs 18

In [6]:
# Your code goes here
num = int(input("Enter any number : "))

temp = num
Sum = 0;

while(temp != 0):

    Sum = Sum + (temp % 10);
    # Remove last digit from temp.
    temp = int(temp / 10)

print("Sum of all digits in", num, "is", Sum)

Enter any number : 2452
Sum of all digits in 2452 is 13


### Task9:
Write a program that accepts a sentence and calculate the number of letters and digits.

Suppose the following input is supplied to the program:

hello world! 123

Then, the output should be:

LETTERS 10

DIGITS 3


In [9]:
# Your code goes here
sentence = input("Enter a Sentence: ")
num_digit = 0
num_letter = 0
for char in sentence:
    if char.isdigit():
        num_digit = num_digit+1
    elif char.isalpha():
        num_letter = num_letter+1
    else:
        pass
print("Letters", num_letter)
print("Digits", num_digit)

Enter a Sentence: Pakistan Zindabad! 786
Letters 16
Digits 3


### Task10: 
You need to trace and make a list of all such numbers from1000 to 3000 in which all the digits are even numbers.

In [10]:
# Your code goes here
values = [] 
for i in range(1000, 3001): 
    if i%2==0: 
        values.append(i) 
    else: 
        pass 
print(values)

[1000, 1002, 1004, 1006, 1008, 1010, 1012, 1014, 1016, 1018, 1020, 1022, 1024, 1026, 1028, 1030, 1032, 1034, 1036, 1038, 1040, 1042, 1044, 1046, 1048, 1050, 1052, 1054, 1056, 1058, 1060, 1062, 1064, 1066, 1068, 1070, 1072, 1074, 1076, 1078, 1080, 1082, 1084, 1086, 1088, 1090, 1092, 1094, 1096, 1098, 1100, 1102, 1104, 1106, 1108, 1110, 1112, 1114, 1116, 1118, 1120, 1122, 1124, 1126, 1128, 1130, 1132, 1134, 1136, 1138, 1140, 1142, 1144, 1146, 1148, 1150, 1152, 1154, 1156, 1158, 1160, 1162, 1164, 1166, 1168, 1170, 1172, 1174, 1176, 1178, 1180, 1182, 1184, 1186, 1188, 1190, 1192, 1194, 1196, 1198, 1200, 1202, 1204, 1206, 1208, 1210, 1212, 1214, 1216, 1218, 1220, 1222, 1224, 1226, 1228, 1230, 1232, 1234, 1236, 1238, 1240, 1242, 1244, 1246, 1248, 1250, 1252, 1254, 1256, 1258, 1260, 1262, 1264, 1266, 1268, 1270, 1272, 1274, 1276, 1278, 1280, 1282, 1284, 1286, 1288, 1290, 1292, 1294, 1296, 1298, 1300, 1302, 1304, 1306, 1308, 1310, 1312, 1314, 1316, 1318, 1320, 1322, 1324, 1326, 1328, 1330, 133

### Task11: 
You need to write a program that counts the number of vowels in a given sentence as input from console. 

Suppose the following input is supplied to the program:

Hello world! Practice makes perfect

Then, the output should be:

a appeared 2 times

e appeared 5 times

i appeared  1 time

o appeared 2 times

u appeared 0 time


In [28]:
# Your code goes here
sentence = input("Enter any Senetence: ")
vowels = ['a','e','i','o','u']
for vowel in vowels:
    print(vowel, "appeard", sentence.count(vowel), "times")

Enter any Senetence: Pakistan Zindabad!
a appeard 4 times
e appeard 0 times
i appeard 2 times
o appeard 0 times
u appeard 0 times


### Task12:
You need to write a program that takes sequence of lines as input and prints the lines after making all characters in the sentence capitalized.

Suppose the following input is supplied to the program:

Hello world

Practice makes perfect

Then, the output should be:

HELLO WORLD
PRACTICE MAKES PERFECT

In [9]:
# Your code goes here
lines = []
while True:
    sentence = input("Enter Lines: ")
    # Leave input blank to stop
    if sentence:
        lines.append(sentence.upper())
    else:
        break
for a_sentence in lines:
    print(a_sentence, end=' ')


Enter Lines: Hello World
Enter Lines: Pakistan Zindabad
Enter Lines: 
HELLO WORLD PAKISTAN ZINDABAD 

### Task13:
You need to write a code that takes a comma separated sequence of words as input and prints the words in a comma-separated sequence after sorting them alphabetically.

Suppose the following input is supplied to the program:

without,hello,bag,world

Then, the output should be:

bag,hello,without,world

In [5]:
# Your code goes here
words = input("Enter Words: ").split(',')
for word in words:
    words.sort()
print(','.join(words))

Enter Words: pan,notebook,book,ink
book,ink,notebook,pan


### Task14:
You need code a program that calculates and prints the value according to the given formula:
P = Square root of [(2 * A * B)/C]

Following are the fixed values of A and B:

A is 50. B is 30.

The values of the literal C should be taken as console input to your program in a comma-separated sequence.

Example:
Let us assume the following comma separated input sequence is given to the program:
100,150,180

The output of the program should be:
18,22,24

In [32]:
# Your code goes here
import math
A=50
B=30
value = []
items=[x for x in input().split(',')]
for C in items:
    value.append((int(round(math.sqrt((2 * A * B)/C)))))

print(','.join(value))

100,150,180


TypeError: unsupported operand type(s) for /: 'int' and 'str'

### Task15:
You need to write a ocde in python that finds all such numbers which are divisible by 7 but are not a multiple of 5,
between 2000 and 3200 (both included).

Print the obtained numbers in a comma-separated sequence on a single line

Hint: modulo operator, and/or operator, for loop /while loop

In [8]:
# Your code goes here
a = []
for i in range(2000, 3001):
    if (i%7==0) and (i%5!=0):
        a.append(str(i))
print(','.join(a))

2002,2009,2016,2023,2037,2044,2051,2058,2072,2079,2086,2093,2107,2114,2121,2128,2142,2149,2156,2163,2177,2184,2191,2198,2212,2219,2226,2233,2247,2254,2261,2268,2282,2289,2296,2303,2317,2324,2331,2338,2352,2359,2366,2373,2387,2394,2401,2408,2422,2429,2436,2443,2457,2464,2471,2478,2492,2499,2506,2513,2527,2534,2541,2548,2562,2569,2576,2583,2597,2604,2611,2618,2632,2639,2646,2653,2667,2674,2681,2688,2702,2709,2716,2723,2737,2744,2751,2758,2772,2779,2786,2793,2807,2814,2821,2828,2842,2849,2856,2863,2877,2884,2891,2898,2912,2919,2926,2933,2947,2954,2961,2968,2982,2989,2996


### Task16:
With a given integral number n, write a program to generate a dictionary that contains (i, i*i) such that is an integral number between 1 and n (both included). and then the program should print the dictionary.

Suppose the following input is supplied to the program:

8

Then, the output should be:

{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64}

In [7]:
# Your code goes here
number = int(input("Enter a Number: "))
d=dict()
for i in range(1,number+1):
    d[i]=i*i
print(d)

Enter a Number: 9
{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64, 9: 81}


### Task17:
You need to write a code which accepts a sequence of comma separated 4 digit binary numbers as its input and then check whether they are divisible by 5 or not. 

The numbers that are divisible by 5 are to be printed in a comma separated sequence.

Example:

0100,0011,1010,1001

Then the output should be:
1010

Reference: 
1) https://www.datacamp.com/.../python-data-type-conversion

2) https://stackoverflow.com/questions/21765779/converting-binary-to-decimal-integer-output?fbclid=IwAR3DD_oiq5pn99IjyxSJgojvB5W6M4JqQtv4EUl_IF_DtmuRXlP-DswcKAA

In [25]:
# Your code goes here
value = []
items=[x for x in input("Enter 4 Binary Numbers: ").split(',')]
for p in items:
    intp = int(p, 2)
    if not intp%5:
        value.append(p)

print(','.join(value))

Enter 4 Binary Numbers: 0100,0011,1010,1001
1010
